modules/irsa/main.tf

```diff
@@ -17,7 +17,7 @@ resource "kubernetes_service_account_v1" "irsa" {
   count = var.create_kubernetes_service_account ? 1 : 0
   metadata {
     name        = var.kubernetes_service_account
-    namespace   = var.kubernetes_namespace
+    namespace   = try(kubernetes_namespace_v1.irsa[count.index].id, var.kubernetes_namespace)
```

Comment:
```suggestion
    namespace   = try(kubernetes_namespace_v1.irsa.metadata[0].name, var.kubernetes_namespace)
```

Reply:
Thanks for the correction. I have commited the codes.